crates/hir_def/src/lib.rs

```diff
@@ -321,11 +329,11 @@ impl_from!(StructId, UnionId, EnumId for AdtId);
 /// A generic param
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub enum GenericParamId {
-    TypeParamId(TypeParamId),
+    TypeParamId(TypeOrConstParamId),
+    ConstParamId(TypeOrConstParamId),
     LifetimeParamId(LifetimeParamId),
-    ConstParamId(ConstParamId),
 }
-impl_from!(TypeParamId, LifetimeParamId, ConstParamId for GenericParamId);
+impl_from!(LifetimeParamId for GenericParamId);
```

Comment:
Why no `TypeOrConstParamId` here?

Reply:
This macro needs variant to have the same name of the type. But here is two variants with the same type, so this macro can't implement `From` trait.

Reply:
Right. Maybe it doesn't make sense to have two variants here if they're then both wrapping `TypeOrConstParamId`? Or we keep `TypeParamId` and `ConstParamId` for this.

Reply:
They were necessary for converting `GenericParamId` to `GenericParam`, so I added `TypeParamId` and `ConstParamId` back.